### PR TITLE
Relax Error Check for 1D cylindrical collection

### PR DESCRIPTION
**Pull request by mgjarrett**
## What is the change?

The `CylindricalComponentsAverageBlockCollection` was recently added: https://github.com/terrapower/armi/pull/1238

This collection is in some ways modeled after the `SlabComponentsAverageBlockCollection`, which is much older. The `_checkComponentConsistency` concept is similar, where each analogous component of each block in the collection is checked for consistency before homogenizing multiple blocks into one representative block. However, the new implementation checked that two components being homogenized into one have _exactly_ the same nuclides. The old implementation just checked key nuclides like U-235, Pu-239, Fe-56, and Na-23 to verify that components were similiar; i.e., structure being merged with structure (Fe-56), fuel with fuel (U-235/Pu-239), coolant with coolant (Na-23).

## Why is the change being made?

The new implementation was excessively picky about what can be homogenized together. Two similar assembly types where one has some extra trace isotopes in the cladding would throw a `ValueError` instead of just graciously being homogenized together.

This PR revises the implementation to make it functionally equivalent to the similar, older method.

Note: using Fe-56 as a proxy for structure and Na-23 for coolant is undesirably SFR-centric, but I'm not going to address that in this PR.

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by keckler** [comment]
@tjayasa FYI, I think your work was one of the motivations for this PR?

File: doc/release/0.2.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ What's new in ARMI
 #. The Spent Fuel Pool (``sfp``) was moved from the ``Core`` out to the ``Reactor``. (`PR#1336 <https://github.com/terrapower/armi/pull/1336>`_)
 #. Removed redundant ``Material.name`` variable. (`PR#1335 <https://github.com/terrapower/armi/pull/1335>`_)
 #. Added SHA1 hashes of XS control files to the welcome text. (`PR#1334 <https://github.com/terrapower/armi/pull/1334>`_)
+#. Cylindrical component block collection less strict about pre-homogenization checks. (`PR#1347 <https://github.com/terrapower/armi/pull/1347>`_)
```
Comment (keckler) [inline doc/release/0.2.rst:14]:
```suggestion
#. Make cylindrical component block collection less strict about pre-homogenization checks. (`PR#1347 <https://github.com/terrapower/armi/pull/1347>`_)
```

**Comment by keckler** [comment]
> Note: using Fe-56 as a proxy for structure and Na-23 for coolant is undesirably SFR-centric, but I'm not going to address that in this PR.

I'd sugget putting a comment in the code saying as much, for future readers.

File: armi/physics/neutronics/crossSectionGroupManager.py [hunk 2]
```diff
@@ -480,7 +481,9 @@ def _checkComponentConsistency(b, repBlock):
 
             theseNucs = set(c.getNuclides())
             thoseNucs = set(repC.getNuclides())
-            diffNucs = theseNucs.symmetric_difference(thoseNucs)
+            diffNucs = theseNucs.symmetric_difference(thoseNucs).intersection(
+                consistentNucs
+            )
```
Comment (keckler) [inline armi/physics/neutronics/crossSectionGroupManager.py:486]: This check is, IMO, pretty opaque and could use a comment. It works, but is one of those "elegant but unreadable" pieces of code.

**Review by keckler — APPROVED** [review]

File: armi/physics/neutronics/crossSectionGroupManager.py [hunk 3]
```diff
@@ -480,7 +481,9 @@ def _checkComponentConsistency(b, repBlock):
 
             theseNucs = set(c.getNuclides())
             thoseNucs = set(repC.getNuclides())
-            diffNucs = theseNucs.symmetric_difference(thoseNucs)
+            diffNucs = theseNucs.symmetric_difference(thoseNucs).intersection(
```
Comment (mgjarrett) [inline armi/physics/neutronics/crossSectionGroupManager.py:484]:
```suggestion
            # check for any differences between which `consistentNucs` the components have
            diffNucs = theseNucs.symmetric_difference(thoseNucs).intersection(
```

File: armi/physics/neutronics/crossSectionGroupManager.py [hunk 4]
```diff
@@ -468,6 +468,7 @@ def _checkComponentConsistency(b, repBlock):
                 f"Blocks {b} and {repBlock} have differing number "
                 f"of components and cannot be homogenized"
             )
+        consistentNucs = {"PU239", "U238", "U235", "U234", "FE56", "NA23", "O16"}
```
Comment (mgjarrett) [inline armi/physics/neutronics/crossSectionGroupManager.py:471]:
```suggestion
        # Using Fe-56 as a proxy for structure and Na-23 as proxy for coolant is undesirably SFR-centric
        # This should be generalized in the future, if possible
        consistentNucs = {"PU239", "U238", "U235", "U234", "FE56", "NA23", "O16"}
```

**Pull request merged**